js/event.ts

```diff
@@ -0,0 +1,282 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types";
+
+export const eventAttributes = new WeakMap();
+
+export class EventInit implements domTypes.EventInit {
+  bubbles = false;
+  cancelable = false;
+  composed = false;
+
+  constructor({bubbles=false, cancelable=false, composed=false} = {}) {
+    this.bubbles = bubbles;
+    this.cancelable = cancelable;
+    this.composed = composed;
+  }
+}
+
+export class Event implements domTypes.Event {
+  // Each event has the following associated flags
+  private _stopPropagationFlag = false;
+  private _stopImmediatePropagationFlag = false;
+  private _canceledFlag = false;
+  private _inPassiveListenerFlag = false;
+
+  // Property for objects on which listeners will be invoked
+  private _path: domTypes.EventPath[] = [];
+
+  constructor(type: string, eventInitDict?: domTypes.EventInit) {
```

Comment:
another nit, you could do:

```suggestion
  constructor(type: string, eventInitDict: domTypes.EventInit = {}) {
```

And then you won't have to do the safety check below, just `eventInitDict.bubbles || false` (still good to do the `||` to ensure a boolean value instead of `undefined`)

Reply:
done